api/src/main/java/org/openmrs/api/impl/OrderServiceImpl.java

```diff
@@ -333,9 +334,9 @@ private Order saveOrderInternal(Order order, OrderContext orderContext) {
 					cal.set(Calendar.MILLISECOND, 0);
 					order.setAutoExpireDate(cal.getTime());
 				}
+				ValidateUtil.validate(order);
```

Comment:
Can you explain why you put this validate method here (inside an if-then block), rather that directly before the dao.saveOrder(order) method, that will get invoked in every case?

Reply:
> Can you explain why you put this validate method here (inside an if-then block), rather that directly before the dao.saveOrder(order) method, that will get invoked in every case?

thanks @mseaton ,just a quick reply to this,i  first added ValidateUtil.validate(order) before the dao.saveOrder(order) method, and it was breaking things,

Reply:
That sounds like something worth investigating.  How was it breaking things?

Reply:
> That sounds like something worth investigating. How was it breaking things?

i am going to re run the change and drop here the logs

Reply:
> That sounds like something worth investigating. How was it breaking things?

hi @mseaton  here are the full logs on putting the validate method  directly before the dao.saveOrder(order), 
  https://pastebin.com/a8LCW0LK

Reply:
Thanks @HerbertYiga - I see the errors.  Have you investigated the cause?

Reply:
> I see the errors. Have you investigated the cause?

@mseaton  well i see the cause of the error is  this line  -> failed to validate with reason: dateActivated: Date activated cannot be before that of the associated encounter